tests/test_sessions_api.py

```diff
@@ -143,14 +116,51 @@ def generate_session_events(self):
                     "status": "errored",
                     "errors": 1,
                     "quantity": 2,
-                    "started": (self.started + timedelta(minutes=14)).timestamp(),
+                    "started": (self.started + timedelta(minutes=24)).timestamp(),
                 },
                 meta,
             ),
         ]
         write_processed_messages(self.storage, events)
 
-    def test_session_aggregation(self):
+    def test_manual_session_aggregation(self):
+        self.generate_manual_session_events()
+        response = self.app.post(
+            "/query",
+            data=json.dumps(
+                {
+                    "dataset": "sessions",
+                    "organization": 1,
+                    "project": 1,
+                    "selected_columns": [
+                        "sessions",
+                        "sessions_errored",
+                        "users",
+                        "users_errored",
+                    ],
+                    "from_date": (self.started - self.skew).isoformat(),
+                    "to_date": (self.started + self.skew).isoformat(),
+                }
+            ),
+        )
+        data = json.loads(response.data)
+        assert response.status_code == 200, response.data
+        assert len(data["data"]) == 1, data
+        assert data["data"][0]["sessions"] == 20
+        assert data["data"][0]["sessions_errored"] == 10
+        assert data["data"][0]["users"] == 2
+        assert data["data"][0]["users_errored"] == 1
+
+    # Since we never select or group by the `bucketed_started`, the granularity
+    # will be ignored basically and we will get the same data.
+    # XXX: When we don’t group by `bucketed_start`, we do get the same data, no
+    # matter if we hit the raw or hourly storage. Right now we can’t really
+    # assert which storage we actually hit.
+    # XXX: Actually applying the parametrize makes snuba return X times the
+    # sessions than are expected. No idea why, as the parametrize down below
+    # works without any problems.
```

Comment:
Since we never select or group by the `bucketed_started`, the granularity will be ignored basically and we will get the same data.

I am not sure I get this. Do you mean in the client you use `started` column instead of bucketed_started? It seems to me you always use the bucketed_started.

Reply:
About returning the same data. That is expected for the test. I think the issue is that we are trying to test something with an end to end test that should be tested with a unit test.

test_discover.py tests something closer to what this should be testing.
https://github.com/getsentry/snuba/blob/master/tests/datasets/test_discover.py#L8

In this case I would write unit tests for `extract_granularity_from_query` passing in query asts and expecting the granularity outside.
Or I would just test the storage selector in a unit test 
https://github.com/getsentry/snuba/pull/1608/commits/3c68e4407de23b70b4bee3a054dd5b44181a49f8#diff-fd0df02e6d42c489b776b141292c684cb53369b0cc5cadcb6a0d15ea3da63b29R202

If you want to test the query processing end to end, this is a safer approach:
https://github.com/getsentry/snuba/blob/a59ebf2c30b1b4e462e82753b714c1ba2b8f6977/tests/datasets/test_events_processing.py

You create and parse a query, then execute the pipeline and inspect the resulting AST pre-execution.